ThunderTable/TableViewController+Reload.swift

```diff
@@ -0,0 +1,112 @@
+    ///
+    /// - Parameters:
+    ///   - row: The row that should be replaced.
+    ///   - otherRow: The row that is replacing the original row.
+    ///   - additionalReloadIndexPaths: Additional index paths that should be reloaded at the same time.
+    ///   - animation: The animation to use when reloading the rows
+    public func replace<T: Row & Equatable>(row: T, with otherRow: Row, reloading additionalReloadIndexPaths: [IndexPath] = [], animation: UITableView.RowAnimation = .none) {
+        
+        guard let indexPath = indexPathFor(row: row) else { return }
+        guard let tableSection = data[indexPath.section] as? TableSection else { return }
+        
+        var rows = tableSection.rows
+        rows[indexPath.row] = otherRow
+        tableSection.rows = rows
+        
+        withoutRedrawing {
+            data[indexPath.section] = tableSection
+        }
+        
+        var indexPaths = [indexPath]
+        indexPaths.append(contentsOf: additionalReloadIndexPaths)
+        tableView.reloadRows(at: indexPaths, with: animation)
+    }
+    
+    /// Replaces multiple rows with replacement rows.
+    ///
+    /// - Note: This relies on the containing section of each row being of class `TableSection` as apposed to any object conforming to `Section`.
+    ///
+    /// - Parameters:
+    ///   - rows: The rows to replace.
+    ///   - otherRows: The rows they should be replaced by.
+    ///   - animation: The row animation to use when replacing
+    public func replace<T: Row & Equatable>(rows: [T], with otherRows: [Row], animation: UITableView.RowAnimation) {
+        
+        guard rows.count == otherRows.count else { return }
+        
+        let replacement: [(Int, IndexPath)] = rows.enumerated().compactMap({
+            guard let indexPath = indexPathFor(row: $0.element) else { return nil }
+            return ($0.offset, indexPath)
+        })
+        
+        replacement.forEach { (index, indexPath) in
+            
+            guard let tableSection = data[indexPath.section] as? TableSection else { return }
+            
+            var rows = tableSection.rows
+            rows[indexPath.row] = otherRows[index]
+            tableSection.rows = rows
+            
+            withoutRedrawing {
+                data[indexPath.section] = tableSection
+            }
+        }
+        
+        tableView.reloadRows(at: replacement.map({ $0.1 }), with: animation)
+    }
+    
+    /// Reloads the cell at the indexPath for a given row.
+    ///
+    /// - Parameters:
+    ///   - row: The row to reload the cell for.
+    ///   - animation: The animation to use when redrawing
+    public func redraw<T: Row & Equatable>(row: T, with animation: UITableView.RowAnimation = .none) {
+        
+        guard let indexPath = indexPathFor(row: row) else { return }
+        tableView.reloadRows(at: [indexPath], with: animation)
```

Comment:
If we are using animation shouldn't we wrap in a `tableView.beginUpdates` - `tableView.endUpdates`? May be wrong there